istresearch/traptor
# Pulse 3138: Update traptor log messages

**BLUF**: updated most log messages to place details in the `extra` section and added an integration test.

**Details**
- Updated log messages allow for better log parsing in ElasticSearch
- Integration test ensures Traptor can obtain rule(s) from Redis and begin collection
- Removed the api keys debug statement
- Added a sample traptor.env file for testing with Docker

**Testing**

Get the code
- `git clone git@github.com:istresearch/traptor.git`
- `git checkout pulse-3138-update-traptor-log-messages`

Set up your environment
- Create and activate a virtual environment (venv or Anaconda)
- Rename `traptor.env.sample` to `traptor.env` and fill in all fields. Some defaults are provided.
- Ensure the Redis database you're using in the `traptor.env` file matches the `docker-compose.yml` file (should be 2)
- Install the requirements: `pip install -r requirements.txt`

Execute the following
- `docker-compose down` if you have any previous containers running
- `docker-compose up --build -d` to build everything
- `docker ps` to check if everything is running

Tail the Logs, Run the Test
- Tail the logs: `tail -f logs/traptor.log`
- Run the integration test file: `python tests/traptor_integration_tests.py`

If everything is working you should see Traptor get it's rules from Redis and begin processing tweets.
